index.d.ts

```diff
@@ -0,0 +1,760 @@
+// Definitions by James George (https://github.com/jamesgeorge007)
+
+export function cli(e: any): any;
+
+export function main(e: any, t: any): any;
+
+interface Config {
+	System: string[];
+	Binaries: string[];
+	Browsers: string[];
+	npmPackages: string | string[];
+	npmGlobalPackages: string[];
+}
+
+interface Options {
+	json: boolean;
+	console: boolean;
+	showNotFound: boolean;
+}
+
+export function run(e: Config, t: Options): any;
```

Comment:
Both `config` and `options` should be this, right?

```
function run(config: Partial<Config>, options: Partial<Options>): any
```

You don't have to specify all config or option keys, you only define what you need.

Reply:
Also, might be nice if we define the actual output of the run function instead of any. I think this should be `string` with any option. AFAIK it's:
  - `showNotFound` => _no effect on return value_
  - `console` => _no effect on return value_
  - `{ json: true }` => json string
  - `{ json: false }` => formatted string